assets/javascripts/overview.js

```diff
@@ -301,17 +303,38 @@ function addComments(form) {
     controls.style.display = 'inline';
     window.addCommentsModal.hide();
   };
-  fetchWithCSRF(form.action, {method: 'POST', body: new FormData(form)})
+
+  let infoTxt =
+    'The comments have been created. <a href="javascript: location.reload()">Reload</a> the page to show changes.';
+  let errTxt = 'The comments could not be added:';
+  if (actionBtn == 'restartAndCommentJobs') {
+    infoTxt = '<a href="javascript: location.reload()">Reload</a> the page to show restarted jobs.';
+    errTx = 'Failed to restart jobs: ';
+  }
+  fetchWithCSRF(reqUrl, {method: 'POST', body: new FormData(form)})
     .then(response => {
       if (!response.ok) throw `Server returned ${response.status}: ${response.statusText}`;
-      addFlash(
-        'info',
-        'The comments have been created. <a href="javascript: location.reload()">Reload</a> the page to show changes.'
-      );
+      addFlash('info', infoTxt);
```

Comment:
This is not good enough. In case of a 200 response not all jobs might have been restarted. In fact, no jobs might have been restarted at all.

When restarting jobs, you might get a 200 response like `{"enforceable":1,"errors":["Job 4339 misses the following mandatory assets: iso\/foo.iso\nEnsure to provide mandatory assets and\/or force re-triggering if necessary."],"warnings",[],"result":[],"test_url":[]}`. The code has to display those errors and allow the user to force the restart if possible. There can also be warnings which also need to be displayed.

I suggest you check how restarting is dealt with in other places of the UI (like the job details page) and see what code you can share.

The code should also read the set of jobs that actually was restarted from the response and then do the commenting only on those jobs (because the comment is likely something like "I am restarting these jobs because of …") so it probably makes no sense to write the comments when the restart didn't actually happen.

Reply:
I apply changes today missing this one. checking.. 
from my understanding, I need to check if `any` job didnt restart and report this. or have a flash for each one which didnt restart?!

Reply:
Don't create a flash for each job that couldn't be restarted. Create one flash message that contains all information. I think we already have code for this because this is not the first place that allows restarting jobs. So check what's already there and share the error handling code.